lib/internal/Magento/Framework/Backup/Db.php

```diff
@@ -45,7 +45,9 @@ public function rollback()
         foreach ($file as $statement) {
             $this->getResourceModel()->runCommand($statement);
         }
-        @unlink($source);
+        if ($this->keepSourceFile() === false) {
```

Comment:
@osrecio so this works when doing a DB backup but does not with the other types of backup. Can you also update the other backup types to include the check for this flag.

Reply:
Made changes for `Fs.php`  and `Ftp.php`